app/server/appsmith-server/src/main/java/com/appsmith/server/authentication/handlers/AuthenticationSuccessHandler.java

```diff
@@ -184,7 +184,8 @@
 
         boolean addFirstTimeExperienceParam = false;
         if (isFromSignup) {
-            if(redirectUrl.endsWith(RedirectHelper.DEFAULT_REDIRECT_URL) && defaultApplication != null) {
+            URI redirectUri = URI.create(redirectUrl);
+            if(redirectUri.getPath().endsWith(RedirectHelper.DEFAULT_REDIRECT_URL) && defaultApplication != null) {
```

Comment:
Going by the docs, the `.getPath()` method may return `null`. So, can we add a null-check, just in case?

```suggestion
            if(redirectUri.getPath() != null && redirectUri.getPath().endsWith(RedirectHelper.DEFAULT_REDIRECT_URL) && defaultApplication != null) {
```